crml/sylo/src/migration.rs

```diff
@@ -71,6 +75,45 @@ decl_module! {
 
 			Ok(())
 		}
+
+		#[weight = SimpleDispatchInfo::FixedOperational(0)]
+		fn migrate_inbox(origin, user_id: T::AccountId, next_index: MessageId, new_messages: Vec<(MessageId, Message)>) -> DispatchResult {
+			Self::ensure_sylo_migrator(origin)?;
+
+			let mut existing_messages = <inbox::Values<T>>::get(&user_id).clone();
+			let (existing_indexes, _): (Vec<MessageId>, Vec<_>) = existing_messages.clone().into_iter().unzip();
+
+			// For repeatability, we update the existing messages that are assumed to be migrated already.
+			for (new_index, new_message) in new_messages {
+				if !existing_indexes.contains(&new_index) {
+					existing_messages.push((new_index, new_message));
+				}
+			}
+
+			ensure!(existing_messages.len() as u32 <= u32::max_value(), Error::<T>::MaxInboxLimitReached);
```

Comment:
`existing_messages.len() as u32` will truncate to `u32`, so this `ensure!` won't ever fail afaict

Reply:
That's true lol. I'll remove this check as I'm sure we don't have this many messages anyway